pkg/cluster/internal/providers/docker/network.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"sigs.k8s.io/kind/pkg/exec"
+)
+
+// TODO: we'll probably allow configuring this
+//
+// however currently picking a single network is equivilant to the previous
+// behavior *except* that we moved from the default bridge to a user defined
+// network because the default bridge is actually special versus any other
+// docker network and lacks the emebdded DNS
+//
+// for now this also makes it easier for apps to join the same network
+const fixedNetworkName = "kind"
+
+// ensureNetwork checks if docker network by name exists, if not it creates it
+func ensureNetwork(name string) error {
+	out, err := exec.Output(exec.Command(
+		"docker", "network", "ls",
+		"--filter=name=^"+name,
```

Comment:
we should use a stricter regex or another way to detect the network

```
docker network ls --filter=name=^k
NETWORK ID          NAME                DRIVER              SCOPE
2054b2e0e513        kind                bridge              local
efc7ccd36802        kindy               bridge              local
```

```
docker network ls --filter=name=^kind$
NETWORK ID          NAME                DRIVER              SCOPE
2054b2e0e513        kind                bridge              local
 aojeagarcia  ~  Downloads  docker network ls --filter=name=^kindy$
NETWORK ID          NAME                DRIVER              SCOPE
efc7ccd36802        kindy               bridge              local
```
or with docker inspect
```
docker inspect kin
[]
Error: No such object: kin
 aojeagarcia  ~  Downloads  1  echo $? 
1
```
```
 docker inspect kind
[
    {
        "Name": "kind",
        "Id": "2054b2e0e513111b323738371afaa6ac80ae22ad4d58e78107f9d4ca6beff7b0",
        "Created": "2020-04-24T15:40:05.557245181+02:00",
...
echo $? 
0
```

Reply:
We should fix the regex.

Reply:
fixed